src/lib/server.ts

```diff
@@ -16,7 +16,13 @@ export async function startServer(app: Application, options: ServerOptions): Pro
 
     // HTTPS - Optional
     if (options.https) {
-      const httpsServer = https.createServer(getHttpsCredentials(options.https.certificate, options.https.privateKey), app);
+      const credentials = getHttpsCredentials(options.https.certificate, options.https.privateKey);
+      let httpsServer;
+      if (options.tlsSecurityOptions) {
```

Comment:
Instead of creating the httpServer here, I would append the secureOptions to the credentials object (rename to something better)
And then create the httpServer with all the options passed

Reply:
Also add a test where the tlsSecurityOptions are passed